pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -225,10 +230,10 @@ def post_dispatch(self) -> None:
 
     def __load_weights_on_main_process(self) -> None:
         model = self.lightning_module
+        assert hasattr(model, "trainer")
```

Comment:
Why this would be needed ?

Reply:
I did this to resolve a TODO right below but i'm with you, maybe it's better to remove the TODO